torch/testing/_internal/common_utils.py

```diff
@@ -5316,3 +5316,46 @@ def repl_frame(m):
         s = re.sub(r"Cannot export model.+\n\n", "", s)
     s = re.sub(r" +$", "", s, flags=re.MULTILINE)
     return s
+
+
+@contextmanager
+def check_leaked_tensors(limit=1):
+    """Wrap around operations you want to ensure are not leaking tensor memory.
+
+    This code intentionally ignores other reference cycles, which can be benign and which we (sadly) have plenty
+    of in pytorch code.  It focuses on any reference cycles that directly or indirectly result holding a Tensor alive,
+    since this is likely a more serious leak than typical python refcycles.
+
+    limit specifies how many tensors to dump debug graphs for (default=1)
+    """
+    try:
+        gc.collect()
+        gc.set_debug(gc.DEBUG_SAVEALL)
+
+        # run the user code, after cleaning any existing refcycles
+        yield
```

Comment:
For testing mentioned below or for better observability you could yield a `Result` class or something similar that has attributes like `has_leak` or `leaked_tensors` and then in the context manager use it like

```python
with check_leaked_tensors() as result:
   ... code
assert result.has_leak == False
```

Reply:
oh, i worked on the fix before i saw your comment. i ended up just yielding the garbage_objs list directly.